### DISCUSSION TITLE
Add latest Unbound to DietPi APT Repository

### DISCUSSION THREAD

**Discussion by Dynamic5912** [MichaIng/DietPi#7019] [Software Request] [answered]
Any chance of adding the latest release of Unbound to the DietPi APT repository?

The latest version on the Stable Debian Repository is 1.17 - version 1.19 is officially released by NLnet and contains several feature and security improvements over 1.17 - this however is only available on the Unstable Debian Repositories.

**Comment by MichaIng** [comment]
If someone creates a build script and workflow for it, like https://github.com/MichaIng/DietPi/tree/dev/.build/software and https://github.com/MichaIng/DietPi/tree/dev/.github/workflows, I would be open to add a package to our repo. But I personally have higher priority tasks, and as always less time than needed to deal with those.

The script does not necessarily need to look like this. Using the Debian sources https://salsa.debian.org/dns-team/unbound/ and debhelper to generate a package that (should) match the one from Debian Sid, just against Debian Bookworm deps/libs, would be fine. Would be even an idea to have a generic workflow+script to build the latest version of any Debian package for any Debian version (based on the container image it runs on).

**Comment by realmax44** [comment]
Hello, I came across this topic through the dietpi forum. As I am very interested in running AGH with Unbound and redis, I would like to contribute, even though I have almost no knowledge. I have created scripts with the help of an AI whose actual functionality I unfortunately cannot judge, but perhaps together we can somehow reach the goal. I have attached the scripts here.
[Scripts.zip](https://github.com/MichaIng/DietPi/files/15074832/Scripts.zip)

**Reply by MichaIng** [reply]
Yeah this uses the Debian sources and debhelper. So we could just need to check whether the default branch is the one we want. We might want to use the lastest `debian/` branch instead: https://salsa.debian.org/dns-team/unbound/-/tree/debian/1.19.2-1

This would need to be wrapped into our `dietpi-software-build` [script](https://github.com/MichaIng/DietPi/blob/dev/.build/software/dietpi-software-build.bash) and [workflow](https://github.com/MichaIng/DietPi/blob/dev/.github/workflows/dietpi-software-build.yml), to allow builds for each arch and Debian version.

**Reply by realmax44** [reply]
Hmm...i don't really understand the wrapping part of your answer. would you please explain, how to do it with the following scripts:

[build.txt](https://github.com/MichaIng/DietPi/files/15076615/build.txt)
[workflow.txt](https://github.com/MichaIng/DietPi/files/15076616/workflow.txt)


then i'll try.

EDIT: something like this?

[wrapped_build.txt](https://github.com/MichaIng/DietPi/files/15078173/wrapped_build.txt)

**Reply by realmax44** [reply]
Sorry, but i'm just learning by doing....

**Reply by MichaIng** [reply]
I answered in your first PR already:
- Add "unbound" to the inputs array https://github.com/MichaIng/DietPi/blob/master/.github/workflows/dietpi-software-build.yml, just behind "ympd".
- Similarly, add "unbound" to the list of supported software names in https://github.com/MichaIng/DietPi/blob/dev/.build/software/dietpi-software-build.bash#L49.
- Add a new directory "unbound" [here](https://github.com/MichaIng/DietPi/tree/dev/.build/software), and the the actual Unbound build script there as "build.bash".

Within the build script, move the resulting `.deb` package(s) to `/tmp`, or better navigate to `/tmp`, before starting the build. I guess it builds all binary packages: https://packages.debian.org/source/bookworm/unbound
I think it would be sufficient to provide `unbound` only, since the other ones have a different dependency on `libunbound`, hence it should be possible to mix most recent `unbound` with the other packages form the Debian repo.

**Comment by MichaIng** [comment, marked as answer]
I merged a workflow for own builds into the just released DietPi v9.15 beta. Will probably give those packages some more time for testing and in case polishing, before shipping them with our APT repository. For now, if you want to test:
1. Not necessarily required, but makes sense to test along with some other not strictly related changes around Unbound, update to DietPi v9.15 beta: #7637
2. Then download and install our package:
    ```sh
    cd /tmp
    wget "https://dietpi.com/downloads/binaries/$G_DISTRO_NAME/unbound_$G_HW_ARCH_NAME.deb"
    apt install ./"unbound_$G_HW_ARCH_NAME.deb"
    ```

All details here: #7632
At best also use that pull request for discussing details, since there are currently multiple issues and discussions split.

**Reply by MichaIng** [reply]
I am closing it here, let's keep discussion in #7632.
